### ISSUE TITLE
[telemetry] Audit queue gauge and audit item counter in ph-oton-audit

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#50]
## Goal

Expose the asynchronous audit queue depth as a ph-telemetry gauge and count audit items by action type, in `ph-oton-audit`.

The queue gauge is the clearest case in ph-oton for the **observable gauge** instrument, which is otherwise unused: an audit queue that silently grows because the backing store has become slow is invisible today until memory runs out.

## Where

**1. `ph-oton-audit/src/main/java/com/helger/photon/audit/AsynchronousAuditor.java`**

```java
public int getQueueLength ()
{
  return m_aRWLock.readLockedInt (m_aCollector::getQueueLength);
}
```

The accessor already exists and is cheap. Wire an observable gauge over it.

The `stop ()` method already logs `"Stopping auditor queue with N items"` when the queue is non-empty at shutdown — the same number that the gauge would have been showing all along.

**2. `ph-oton-audit/src/main/java/com/helger/photon/audit/AbstractAuditor.java`**

```java
public void createAuditItem (@NonNull final EAuditActionType eActionType, ...)
protected abstract void handleAuditItem (@NonNull IAuditItem aAuditItem);
```

`createAuditItem` is the central entry point shared by all auditor implementations (`AsynchronousAuditor`, `LoggingAuditor`, `AuditorJDBC`, `DoNothingAuditor`). Put the counter there so every backend is covered by one increment.

## What to emit

New `com.helger.photon.audit.CAuditTelemetry`:

| Constant | Value |
|---|---|
| `METRIC_AUDIT_ITEMS` | `photon.audit.items` |
| `METRIC_AUDIT_QUEUE_LENGTH` | `photon.audit.queue.length` |
| `ATTR_AUDIT_ACTION_TYPE` | `photon.audit.action.type` |
| `ATTR_AUDIT_SUCCESS` | `photon.audit.success` |
| `UNIT_ITEM` | `{item}` |

- **Counter** `photon.audit.items`, attributes `action.type` = the `EAuditActionType` value and `success`. `EAuditActionType` has exactly five values (`CREATE`, `MODIFY`, `DELETE`, `UNDELETE`, `EXECUTE`), so it is an ideal dimension. Use the enum's ID (`"create"`, `"modify"`, …) — it implements `IHasID <String>`; verify the accessor before use.
- **Gauge** `photon.audit.queue.length` over `AsynchronousAuditor.getQueueLength ()`, no attributes.

## Cardinality — the important exclusion

Do **not** put the audit *action string* or the audit *arguments* on any attribute. `IAuditActionStringProvider` produces free-form text containing object IDs, user IDs and field values — unbounded, and frequently personal data. Only the action **type** enum is safe.

The user ID from `ICurrentUserIDProvider` must likewise never become a metric attribute.

## Gauge lifecycle caveat

`TelemetryMetrics.gauge (...)` returns a `Closeable` `ITelemetryGauge`. `AsynchronousAuditor` is an instance with an explicit `stop ()` method (and a `SimpleReadWriteLock` guarding its collector), so the gauge must be created in the constructor and closed in `stop ()` — **not** held in a `static final` field, or a stopped auditor will be kept alive by the gauge callback and a restarted one will double-report.

The counter in `AbstractAuditor`, by contrast, is stateless and *should* be a `static final` in a `AuditMetrics` holder class, as in `LongRunningJobMetrics`.

## POM

`ph-oton-audit/pom.xml` needs a new dependency (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

## Acceptance criteria

- [ ] Counter fires once per `createAuditItem (...)` regardless of the concrete auditor implementation.
- [ ] Gauge reports `getQueueLength ()` and is closed in `AsynchronousAuditor.stop ()`.
- [ ] No audit action string, audit argument or user ID on any attribute.
- [ ] Unit test with recording SPIs installed via `TelemetryMetrics.install (...)` and reset to `null` in `@After`.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.6.0**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-audit/.../audit/CAuditTelemetry.java` | new, public - metric / attribute names |
| `ph-oton-audit/.../audit/AuditMetrics.java` | new, public - counter + gauge factory |
| `ph-oton-audit/.../audit/AuditTelemetry.java` | new, package-private emitter |
| `ph-oton-audit/.../audit/AbstractAuditor.java` | modified - one counter increment |
| `ph-oton-audit/.../audit/AsynchronousAuditor.java` | modified - gauge created in the ctor, closed in `stop ()` |
| `ph-oton-audit/pom.xml` | `com.helger.telemetry:ph-telemetry` added (no `<version>`, BOM) |
| `ph-oton-audit/src/test/.../AuditTelemetryTest.java` + `RecordingTelemetryMeterSPI.java` + service file | new, 2 tests |

### Emission

* Counter `photon.audit.items` in `AbstractAuditor.createAuditItem (...)`, immediately before `handleAuditItem (...)`. That is the single entry point shared by `AsynchronousAuditor`, `LoggingAuditor`, `AuditorJDBC` and `DoNothingAuditor`, so one increment covers every backend.
* Observable gauge `photon.audit.queue.length` over `AsynchronousAuditor.getQueueLength ()`, no attributes.

`EAuditActionType` implements `IHasID <String>`, so the attribute value is `eActionType.getID ()` - `create`, `modify`, `delete`, `undelete`, `execute`. Five values, ideal dimension.

### Gauge lifecycle

As required, an instance field and not a `static final`:

```java
// Bound to the life time of this auditor - must not be a static field
private final ITelemetryGauge m_aQueueLengthGauge;
```

created at the end of the constructor and closed inside the write-locked section of `stop ()`, right after the existing "Stopping auditor queue with N items" log line. `AuditTelemetryTest.testQueueLengthGauge` asserts that the gauge exists while the auditor runs and is gone after `stop ()`, so a stopped auditor is not kept alive by the callback and a restarted one cannot double-report.

### Cardinality - the important exclusion

Only `photon.audit.action.type` and `photon.audit.success` are attributes. The audit **action string** and the **arguments** come from an `IAuditActionStringProvider` and contain object IDs, user IDs and field values; the user ID comes from `ICurrentUserIDProvider`. None of them is used. The test passes `"userid"` as the current user ID and free-text arguments, then asserts the attribute map has exactly 2 entries and does not contain that user ID.

### Test infrastructure

`LoggingAuditorTest` and `AsynchronousAuditorTest` both create audit items and sort before `AuditTelemetryTest`, so they would bind `AuditMetrics.AUDIT_ITEMS` to the no-op meter before any `TelemetryMetrics.install (...)` could take effect. The recording meter is therefore registered through the SPI mechanism - `src/test/resources/META-INF/services/com.helger.telemetry.ITelemetryMeterSPI` - exactly as in ph-oton-security. `SPITest` passes with the new service file.

### Verification

Whole ph-oton-audit suite: 9/9 green, including the pre-existing auditor tests and `SPITest`. Full reactor `mvn install` over all 23 modules: BUILD SUCCESS. `README.md` news bullet added under `v10.6.0 - work in progress`.
